cli/command/registry/login.go

```diff
@@ -47,6 +51,34 @@ func runLogin(dockerCli command.Cli, opts loginOptions) error {
 	ctx := context.Background()
 	clnt := dockerCli.Client()
 
+	if opts.password != "" {
+		fmt.Fprintf(os.Stderr, "Using --password via the CLI is insecure. Please use --password-file.\n")
+	}
+
+	if opts.passwordFile != "" {
```

Comment:
We should also support reading from stdin if `passwordFile == "-"`. There is code to handle that from an arg in other commands. I'm not sure if we have a function for it yet, or if maybe we could extract one.